tests/Test/Xrefcheck/TrailingSlashSpec.hs

```diff
@@ -0,0 +1,45 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.TrailingSlashSpec where
+
+import qualified Unsafe
+
+import Test.Hspec (Spec, describe, it, runIO)
+import Test.QuickCheck ((===))
+
+import Xrefcheck.Config
+import Xrefcheck.Core
+import Xrefcheck.Progress
+import Xrefcheck.Scan
+import Xrefcheck.Scanners.Markdown
+import Xrefcheck.Verify
+
+spec :: Spec
+spec = do
+  describe "Trailing forward slash detection" $ do
+    let config = defConfig GitHub
+    let format = specificFormatsSupport [markdownSupport (scMarkdown (cScanners config))]
+    forM_ roots $ \root -> do
+      repoInfo <- runIO $ allowRewrite False $ \rw ->
+        gatherRepoInfo rw format TraversalConfig{ tcIgnored = [] } root
+      res <- runIO $ allowRewrite False $ \rw ->
+        verifyRepo rw (cVerification config) FullMode root repoInfo
+      it ("Analysis results for the root with" <>
+          (if Unsafe.last root == '/' then "" else "out") <>
+          " the trailing forward slash should not contain the FileDoesNotExist error") $ do
+        let satisfactoryCase = maybe True (not . any checkFileDoesNotExist) $ verifyErrors res
+        satisfactoryCase === True
```

Comment:
Comparing two bools, like `bool === True`, usually leads to less-than-ideal error messages.

For example, if we expected files `a`, `b` and `c` to exist, but `b` didn't exist, the error message in the test logs will simply say:

> `Expected 'True', got 'False'`.

Ideally, the error message should say something like: 

> Expected all filepaths to be valid, but this filepath does not exist: b